### PR TITLE
Ensure STC is installed with Pyccel

### LINKED ISSUE #1654: Ensure STC is installed with Pyccel
### Relevant Discussion

https://github.com/pyccel/pyccel/discussions/1053#discussioncomment-7290234

### Describe the feature
In order to add support for containers we need to include our new dependencies in Pyccel and ensure that they are correctly installed. In C we will use https://github.com/Stclib/STC/ . We need to make sure that the necessary files for compilation are packaged with Pyccel.

### Proposed Solution
Add https://github.com/Stclib/STC/ as a git submodule. Update the `pyproject.toml` file to ensure that the necessary files from STC are installed with Pyccel. We do not need to install the docs from the library.

There may need to be an investigation to find the most appropriate way to handle this. I think it would be a good idea to create a folder called `extensions` at the root and keep dependencies here. It is not clear how to map the include and src folders to the Pyccel folder from there. The solution will involve artifacts and may require symbolic links

### PR DESCRIPTION
**Pull request by mustapha-belbiad**
Fix #1654:

- Add https://github.com/Stclib/STC as a git submodule.
- Update file `pyproject.toml` to include essential files from STC.
- Implement an action in `check_for_stc` to verify STC installation.

### CODE REVIEW

File: .github/actions/check_for_stc/action.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  using: "composite"
+  steps:
+    - name: check stc installation
+      shell: python
+      run: |
+        from extensions import STC
+        from os.path import exists, join
+        pyccel_loc = stc.__path__[0]
+        print(pyccel_loc)
+        assert(exists(join(pyccel_loc,'include')))
```
Comment (EmilyBourne) [inline .github/actions/check_for_stc/action.yml:13]: You need to ensure that this action is called from a workflow. I think it makes sense in the workflows which check the wheels

Comment (EmilyBourne) [reply .github/actions/check_for_stc/action.yml:13]: Sorry. I'll leave you to it and only look when you request a review ;)

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -44,6 +44,10 @@ jobs:
           ref: ${{ env.COMMIT }}
           repository: ${{ env.PR_REPO }}
           fetch-depth: 0
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          submodules: true
```
Comment (EmilyBourne) [inline .github/workflows/docs.yml:50]: Why did you add a new checkout action here?

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -61,6 +61,10 @@ jobs:
         with:
           ref: ${{ env.COMMIT }}
           repository: ${{ env.PR_REPO }}
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          submodules: true
```
Comment (EmilyBourne) [inline .github/workflows/linux.yml:67]: Does this workflow checkout Pyccel twice?

File: .github/workflows/windows.yml [hunk 4]
```diff
@@ -57,6 +57,10 @@ jobs:
         with:
           ref: ${{ env.COMMIT }}
           repository: ${{ env.PR_REPO }}
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
```
Comment (EmilyBourne) [inline .github/workflows/windows.yml:61]: You don't need to specify the minor version of the checkout action

**Review by EmilyBourne — CHANGES_REQUESTED** [review]
Don't forget the other actions that are only run on the devel branch

File: .github/workflows/docs.yml (hunk 2 above)
Comment (EmilyBourne) [reply .github/workflows/docs.yml:50]: That explains why you added:
```
          submodules: true
```
Not why you added a new:
```
      - name: Checkout
        uses: actions/checkout@v4.1.1
```

File: .github/workflows/linux.yml (hunk 3 above)
Comment (mustapha-belbiad) [reply .github/workflows/linux.yml:67]: I think each workflow needs to check out submodules

Comment (EmilyBourne) [reply .github/workflows/linux.yml:67]: I agree. That is not my question

Comment (EmilyBourne) [reply .github/workflows/linux.yml:67]: These lines check out more than just submodules

File: .github/workflows/docs.yml [hunk 5]
```diff
@@ -44,6 +44,7 @@ jobs:
           ref: ${{ env.COMMIT }}
           repository: ${{ env.PR_REPO }}
           fetch-depth: 0
+          submodules: true
```
Comment (EmilyBourne) [inline .github/workflows/docs.yml:47]: There are 2 checkout commands in this file

Comment (mustapha-belbiad) [reply .github/workflows/docs.yml:48]: Fixed :  e721e3d

File: .github/workflows/coverage.yml [hunk 6]
```diff
@@ -44,6 +44,7 @@ jobs:
         with:
           ref: ${{ env.COMMIT }}
           repository: ${{ env.PR_REPO }}
+          submodules: true
```
Comment (EmilyBourne) [inline .github/workflows/coverage.yml:47]: There are 2 checkout commands in this file

File: .github/workflows/docs.yml [hunk 7]
```diff
@@ -38,12 +38,14 @@ jobs:
         with:
           ref: ${{ inputs.base }}
           path: base
+           submodules: true
```
Comment (EmilyBourne) [inline .github/workflows/docs.yml:41]: Indentation error

Comment (mustapha-belbiad) [reply .github/workflows/docs.yml]: fixed : f28f58c

File: .github/workflows/pickle_wheel.yml [hunk 8]
```diff
@@ -78,8 +79,11 @@ jobs:
       - name: Check for existence of pickled files
         id: pickle_check
         uses: ./.github/actions/check_for_pickled
+      - name: Check for STC installation
+        id: stc_check
+        uses: ./.github/actions/check_for_stc
       - name: "Post completed"
```
[the remark this answers is not in the document]
Comment (mustapha-belbiad) [reply .github/workflows/pickle_wheel.yml:85]: You're right, i will add it in my next commit.

[the remark this answers is not in the document]
Comment (mustapha-belbiad) [reply .github/workflows/pickle_wheel.yml:85]: Fixed : b9e1d08

File: .github/workflows/comment_bot.yml [hunk 9]
```diff
@@ -12,6 +12,7 @@ jobs:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
+          submodules: true
```
Comment (EmilyBourne) [inline .github/workflows/comment_bot.yml:15]: The bot doesn't need access to the submodules

File: .github/workflows/clean_up.yml [hunk 10]
```diff
@@ -13,6 +13,7 @@ jobs:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
+          submodules: true
```
Comment (EmilyBourne) [inline .github/workflows/clean_up.yml:16]: The bot doesn't need access to the submodules

File: .github/workflows/pickle.yml [hunk 11]
```diff
@@ -87,6 +88,9 @@ jobs:
       - name: Check for existence of pickled files
         id: pickle_check
         uses: ./.github/actions/check_for_pickled
+      - name: Check for STC installation
+        id: stc_check
+        uses: ./.github/actions/check_for_stc
```
Comment (EmilyBourne) [inline .github/workflows/pickle.yml:93]: Should be added in status checks

File: .github/workflows/prepare_review_stage.yml [hunk 12]
```diff
@@ -13,6 +13,7 @@ jobs:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
+          submodules: true
```
Comment (EmilyBourne) [inline .github/workflows/prepare_review_stage.yml:16]: The bot doesn't need access to the submodules

File: .github/workflows/pickle.yml (hunk 11 above)
Comment (mustapha-belbiad) [reply .github/workflows/pickle.yml:93]: Fixed :[dc3fcce

File: .github/workflows/prepare_review_stage.yml (hunk 12 above)
Comment (mustapha-belbiad) [reply .github/workflows/prepare_review_stage.yml]: Fixed : f406515

File: .github/workflows/clean_up.yml (hunk 10 above)
Comment (mustapha-belbiad) [reply .github/workflows/clean_up.yml]: Fixed : f406515

File: .github/workflows/comment_bot.yml (hunk 9 above)
Comment (mustapha-belbiad) [reply .github/workflows/comment_bot.yml]: Fixed : f406515

**Review by EmilyBourne — DISMISSED** [review]
This looks good to me. I will trigger a complete test suite

File: pyproject.toml [hunk 13]
```diff
@@ -52,3 +52,6 @@ artifacts = [
 [tool.hatch.build.targets.wheel.hooks.custom]
 path = "install_scripts/hook.py"
 require-runtime-dependencies = true
+
+[tool.hatch.build.targets.wheel.force-include]
+"./extensions/STC/include/stc" = "./pyccel/stc"
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyproject.toml:57]: See https://hatch.pypa.io/1.9/plugins/builder/wheel/

> A [wheel](https://packaging.python.org/specifications/binary-distribution-format/) is a binary distribution of a Python package that can be installed directly into an environment.

I suspect what is needed is sdist: https://hatch.pypa.io/1.9/plugins/builder/sdist/

> A source distribution, or sdist, is an archive of Python "source code". Although largely unspecified, by convention it should include everything that is required to build a [wheel](https://hatch.pypa.io/1.9/plugins/builder/wheel/) without making network requests.

**Comment by EmilyBourne** [comment]
@yguclu The implementation that we suggested where we install into `/home/emily/.local/lib/python3.10/site-packages/pyccel/stc` doesn't work in editable mode. In editable mode any files that can be modified aren't moved and the only symbolic links that is created is in `/home/emily/.local/lib/python3.10/site-packages/`.

I suggest a slightly different location to work round this. E.g. `/home/emily/.local/lib/python3.10/site-packages/pyccel_externals`

So far we have only worked out how to copy these additional files, not create an extra link.
This means that when we install in editable mode the external files are not editable. But they can be easily accessed.

So in an editable installation we would have at the start:
```
/home/emily/Code/pyccel/pyccel
/home/emily/Code/pyccel/externals/STC/include/stc
```
and after installation:
```
/home/emily/Code/pyccel/pyccel
/home/emily/.local/lib/python3.10/site-packages/pyccel_externals/stc
```
And in a non editable installation we would have at the start:
```
/home/emily/Code/pyccel/pyccel
/home/emily/Code/pyccel/externals/STC/include/stc
```
and after installation:
```
/home/emily/.local/lib/python3.10/site-packages/pyccel
/home/emily/.local/lib/python3.10/site-packages/pyccel_externals/stc
```

What do you think? In my opinion a fully editable installation would be better but the external files should rarely be modified so I think this solution would be sufficient for now

**Comment by EmilyBourne** [comment]
@mustapha-belbiad What version of STC did you pin to? A specific release or just the current state of the master branch?

**Comment by EmilyBourne** [comment]
> So far we have only worked out how to copy these additional files, not create an extra link.

After some more googling it seems that this is roughly equivalent to installing 2 projects from 1 `pyproject.toml` which is not advised or fully supported (at least by hatch). So either we need to accept that files that we should not be modifying (except to update the version) cannot be modified. Or we need to put the externals/ folder in pyccel/ and we cannot change the path (although we can avoid installing the README/docs/etc). Which is preferred? @yguclu @mustapha-belbiad

File: pyproject.toml [hunk 14]
```diff
@@ -54,7 +54,7 @@ path = "install_scripts/hook.py"
 require-runtime-dependencies = true
 
 [tool.hatch.build.targets.wheel.force-include]
-"./pyccel/extensions/STC/include/stc" = "./pyccel/extensions/stc"
+"./pyccel/extensions/STC/include/stc" = "./pyccel/stc"
 
 [tool.hatch.build.targets.wheel.force_include_editable]
-"./pyccel/extensions/STC/include/stc" = "./pyccel/extensions/stc"
\ No newline at end of file
+"./pyccel/extensions/STC/include/stc" = "./pyccel/stc"
```
Comment (EmilyBourne) [inline pyproject.toml]: Have you tried with `sdist` instead of `wheel`? I don't think a wheel is created for an editable installation

Comment (mustapha-belbiad) [reply pyproject.toml]: Yes, unfortunately it didn't work. I'll need to look into it to identify the problem.

File: .github/actions/check_for_stc/action.yml [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+  using: "composite"
+  steps:
+    - name: check stc installation
+      shell: python
+      run: |
+        import pyccel
+        from os.path import exists, join
+        pyccel_loc = pyccel.__path__[0]
+        print(pyccel_loc)
+        assert(exists(join(pyccel_loc,'extensions/STC/include/stc')))
```
Comment (EmilyBourne) [inline .github/actions/check_for_stc/action.yml:13]: Please can you add a few lines to this action to make sure that we don't install the docs. I don't want them creeping back in in the future.

File: docs/installation.md [hunk 16]
```diff
@@ -203,6 +203,7 @@ for a system-wide installation.
     ```sh
     git clone git@github.com:pyccel/pyccel.git
     cd pyccel
+    git submodule update --init --recursive
```
Comment (EmilyBourne) [inline docs/installation.md:206]:
```suggestion
    git clone --recurse-submodules git@github.com:pyccel/pyccel.git
    cd pyccel
```

File: docs/installation.md [hunk 17]
```diff
@@ -211,6 +212,7 @@ for a system-wide installation.
...
```
Comment (EmilyBourne) [inline docs/installation.md:215]: See above

File: docs/installation.md (hunk 16 above)
Comment (mustapha-belbiad) [reply docs/installation.md]: fixed : f76c559

File: docs/installation.md (hunk 17 above)
Comment (mustapha-belbiad) [reply docs/installation.md]: fixed : f76c559

File: .github/actions/check_for_stc/action.yml (hunk 15 above)
Comment (mustapha-belbiad) [reply .github/actions/check_for_stc/action.yml:13]: Fixed : dd1abd8

**Review by EmilyBourne — APPROVED** [review]
This looks good to me now. Thanks for taking care of this

File: .github/actions/check_for_stc/action.yml [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+        assert(not exists(join(pyccel_loc,'extensions/STC/docs')))
```
Comment (EmilyBourne) [inline .github/actions/check_for_stc/action.yml:14]: Don't you need an if condition here so it doesn't complain in editable mode?

Comment (EmilyBourne) [reply .github/actions/check_for_stc/action.yml:14]: Answer, yes:
https://github.com/pyccel/pyccel/actions/runs/8468392946/job/23201312501

Comment (mustapha-belbiad) [reply .github/actions/check_for_stc/action.yml:14]: you're right, I will add it in my next commit.

Comment (mustapha-belbiad) [reply .github/actions/check_for_stc/action.yml]: I added a condition her :  787aeb9. Let me know if it's the right way to solve this problem.

Comment (EmilyBourne) [reply .github/actions/check_for_stc/action.yml]: I'm not sure this is the right way to do this. A user installation (`--user`) is writable but is not necessarily editable. I think the simplest is to parametrise the test

Comment (EmilyBourne) [reply .github/actions/check_for_stc/action.yml]: The link tells you what to put in the action.yml file. Maybe you can look at our `post_completed` action for an example

File: .github/actions/check_for_stc/action.yml [hunk 19]
```diff
@@ -0,0 +1,18 @@
+name: 'Ensure STC is installed with Pyccel'
+inputs:
+  editable:
+    description: 'editable installation'
```
Comment (EmilyBourne) [inline .github/actions/check_for_stc/action.yml:4]: It might be easier to add a default value for this

Comment (mustapha-belbiad) [reply .github/actions/check_for_stc/action.yml:4]: Okay, but what should be the default value? How about using `default: '-e'` or `editable`?

Comment (EmilyBourne) [reply .github/actions/check_for_stc/action.yml:4]: I think the default should be the non-editable case as this is used more often

File: .github/actions/check_for_stc/action.yml [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+    - name: check stc installation
+      shell: python
+      run: |
+        import pyccel, os
+        from os.path import exists, join
+        pyccel_loc = pyccel.__path__[0]
+        print(pyccel_loc)
+        assert(exists(join(pyccel_loc,'extensions/STC/include/stc')))
+        tmp = "${{ inputs.editable }}"
+        if tmp is None:
```
[the remark this answers is not in the document]
Comment (mustapha-belbiad) [reply .github/actions/check_for_stc/action.yml]: Fixed : 8007dce

File: .github/actions/check_for_stc/action.yml [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+    - name: check stc installation
+      shell: python
+      run: |
+        import pyccel, os
+        from os.path import exists, join
+        pyccel_loc = pyccel.__path__[0]
+        print(pyccel_loc)
+        assert(exists(join(pyccel_loc,'extensions/STC/include/stc')))
+        if "${{ inputs.editable }}" == "false":
+          assert(not exists(join(pyccel_loc,'extensions/STC/docs')))
```
Comment (EmilyBourne) [inline .github/actions/check_for_stc/action.yml:18]: Why use strings here? Instead of:
```
if "false" == "false":
    assert ...
```
You could simply do:
```
if true:
```
You may need to change the defaults/the argument for this

File: .github/workflows/pickle.yml [hunk 22]
```diff
@@ -87,8 +88,13 @@ jobs:
...
+        with:
+          editable: 'False'
```
Comment (EmilyBourne) [inline .github/workflows/pickle.yml:95]: This value needs to depend on "editable_string". Currently it is False even if the installation is editable

Comment (mustapha-belbiad) [reply .github/workflows/pickle.yml]: Fixed : 12dfa29

File: .github/actions/check_for_stc/action.yml [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+    - name: check stc installation
+      shell: python
+      run: |
+        import pyccel, os
+        from os.path import exists, join
+        pyccel_loc = pyccel.__path__[0]
+        print(pyccel_loc)
+        assert(exists(join(pyccel_loc,'extensions/STC/include/stc')))
+        if ${{ inputs.editable }}:
+          assert(not exists(join(pyccel_loc,'extensions/STC/docs')))
```
[the remark this answers is not in the document]
Comment (mustapha-belbiad) [reply .github/actions/check_for_stc/action.yml]: Fixed : 12dfa29

File: .github/workflows/pickle.yml [hunk 24]
```diff
@@ -87,8 +88,13 @@ jobs:
       - name: Check for existence of pickled files
         id: pickle_check
         uses: ./.github/actions/check_for_pickled
+      - name: Check for STC installation
+        id: stc_check
+        uses: ./.github/actions/check_for_stc
+        with:
+          not_editable: 'false'
```
Comment (EmilyBourne) [inline .github/workflows/pickle.yml:95]: This workflow can be editable or non-editable depending on the contents of `${{ inputs.editable_string }}`. You will need to check this value. If you specify the type of the argument (boolean) in the `check_for_stc` action then you should be able to pass something like `${{ inputs.editable_string == '-e' }}` here

File: .github/actions/check_for_stc/action.yml [hunk 25]
```diff
@@ -13,6 +13,7 @@ runs:
         from os.path import exists, join
         pyccel_loc = pyccel.__path__[0]
         print(pyccel_loc)
+        print("==========================================------------------> ${{ inputs.editable_string }}")
```
Comment (EmilyBourne) [inline .github/actions/check_for_stc/action.yml]: You don't have access to this here. It is defined in one (but not all) of the workflows that calls this action. It is not an input to this action

File: .github/workflows/pickle.yml (hunk 24 above)
Comment (EmilyBourne) [reply .github/workflows/pickle.yml:95]: The fix for this should be done in this file

Comment (EmilyBourne) [reply .github/workflows/pickle.yml:95]: >  If you specify the type of the argument

Ah unfortunately this only seems possible with a workflow

Comment (EmilyBourne) [reply .github/workflows/pickle.yml:95]: This may also be useful: https://docs.github.com/en/actions/learn-github-actions/expressions#example

File: .github/actions/check_for_stc/action.yml [hunk 26]
```diff
@@ -0,0 +1,18 @@
...
+    - name: check stc installation
+      shell: python
+      run: |
+        import pyccel, os
+        from os.path import exists, join
+        pyccel_loc = pyccel.__path__[0]
+        print(pyccel_loc)
+        print("${{ inputs.not_editable }}")
+        assert(exists(join(pyccel_loc,'extensions/STC/include/stc')))
+        if "${{ inputs.not_editable }}" != "-e":
```
Comment (EmilyBourne) [inline .github/actions/check_for_stc/action.yml:17]: Ideally I don't think you should mention `-e` in this file as there are probably other ways of handling editable installations so it would be good to use something more general like you had before

File: .github/workflows/pickle.yml [hunk 27]
```diff
@@ -92,7 +92,7 @@ jobs:
         id: stc_check
         uses: ./.github/actions/check_for_stc
         with:
-          not_editable: "${{ matrix.editable_string }}"
+          not_editable: "${{ matrix.editable_string == "-e" && "False" || "True"}}"
```
Comment (EmilyBourne) [inline .github/workflows/pickle.yml:95]: https://docs.github.com/en/actions/learn-github-actions/expressions#literals
> You don't need to enclose strings in ${{ and }}. However, if you do, you must use single quotes (') around the string. To use a literal single quote, escape the literal single quote using an additional single quote (''). Wrapping with double quotes (") will throw an error.

File: .github/workflows/docs.yml [hunk 28]
```diff
@@ -75,8 +77,8 @@ jobs:
         id: doc_coverage
         run: |
           echo "${{ github.event.pull_request.base.sha }}"
-          docstr-coverage --skip-magic --skip-init --fail-under=0 --docstr-ignore-file=base/.docstr_coverage base/pyccel base/ci_tools 2>&1 | tee base_cov
-          docstr-coverage --skip-magic --skip-init --fail-under=0 --docstr-ignore-file=compare/.docstr_coverage compare/pyccel compare/ci_tools 2>&1 | tee compare_cov
+          docstr-coverage --skip-magic --skip-init --fail-under=0  --exclude=base/pycce/extensions --docstr-ignore-file=base/.docstr_coverage base/pyccel base/ci_tools 2>&1 | tee base_cov
+          docstr-coverage --skip-magic --skip-init --fail-under=0 --exclude=compare/pycce/extensions --docstr-ignore-file=compare/.docstr_coverage compare/pyccel compare/ci_tools 2>&1 | tee compare_cov
```
Comment (EmilyBourne) [inline .github/workflows/docs.yml:81]:
```suggestion
          docstr-coverage --skip-magic --skip-init --fail-under=0 --exclude=base/pyccel/extensions --docstr-ignore-file=base/.docstr_coverage base/pyccel base/ci_tools 2>&1 | tee base_cov
          docstr-coverage --skip-magic --skip-init --fail-under=0 --exclude=compare/pyccel/extensions --docstr-ignore-file=compare/.docstr_coverage compare/pyccel compare/ci_tools 2>&1 | tee compare_cov
```

Comment (mustapha-belbiad) [reply .github/workflows/docs.yml]: Fixed : e0c4db5

File: .github/workflows/docs.yml [hunk 29]
```diff
@@ -75,8 +77,8 @@ jobs:
         id: doc_coverage
         run: |
           echo "${{ github.event.pull_request.base.sha }}"
-          docstr-coverage --skip-magic --skip-init --fail-under=0 --docstr-ignore-file=base/.docstr_coverage base/pyccel base/ci_tools 2>&1 | tee base_cov
-          docstr-coverage --skip-magic --skip-init --fail-under=0 --docstr-ignore-file=compare/.docstr_coverage compare/pyccel compare/ci_tools 2>&1 | tee compare_cov
+          docstr-coverage --skip-magic --skip-init --fail-under=0 --exclude=base/pyccel/extensions --docstr-ignore-file=base/.docstr_coverage base/pyccel base/ci_tools 2>&1 | tee base_cov
+          docstr-coverage --skip-magic --skip-init --fail-under=0 --exclude=compare/pyccel/extensions --docstr-ignore-file=compare/.docstr_coverage compare/pyccel compare/ci_tools 2>&1 | tee compare_cov
```
Comment (EmilyBourne) [inline .github/workflows/docs.yml:81]: Maybe try:
```suggestion
          docstr-coverage --skip-magic --skip-init --fail-under=0 --exclude=".*base/pyccel/extensions" --docstr-ignore-file=base/.docstr_coverage base/pyccel base/ci_tools 2>&1 | tee base_cov
          docstr-coverage --skip-magic --skip-init --fail-under=0 --exclude=".*compare/pyccel/extensions" --docstr-ignore-file=compare/.docstr_coverage compare/pyccel compare/ci_tools 2>&1 | tee compare_cov
```
so the regex matches the absolute path?

**Review by EmilyBourne — APPROVED** [review]

File: pyproject.toml [hunk 30]
```diff
@@ -48,7 +48,23 @@ path = "pyccel/version.py"
...
+  "*.pyccel",
+  "pyccel/stdlib/*/*.h",
+  "pyccel/stdlib/*/*.f90",
+  "*.tx",
+  "*.c"
+]
+exclude = [
+    "misc",
+    "src"
+]
```
Comment (yguclu) [inline pyproject.toml:64]: Can you please explain how you have determined what to list under `include` and `exclude`?

This is not totally clear to me. For example, I expected that we should write "pyccel/extensions/STC/misc" instead of just "misc"...

**Review by yguclu — COMMENTED** [review]
Good job! I just have a couple of comments.

File: pyproject.toml (hunk 30 above)
Comment (yguclu) [reply pyproject.toml:64]: @EmilyBourne Do you happen to know the answer?

Comment (EmilyBourne) [reply pyproject.toml:64]: I'm not sure but I can take a guess. The `exclude` folder also doesn't list everything in STC (e.g. `docs` is not listed) but seems to be sufficient to exclude these folders.

I think that once an `include` section is specified it is used to define all files that should be included using this as a regex. Given that the regex for c files is simply `*.c` this is probably pulling in `misc` (pyccel/extensions/STC/misc/)  which contains `*.c` files and `src` (pyccel/extensions/STC/src/) which contains both `*.c` and  `*.py` files.

I agree that it would be better to use more explicit paths for this though as it should avoid several of these conflicts. Do you want me to fix it?

Comment (EmilyBourne) [reply pyproject.toml:64]: I have run a couple of tests and this explanation seems to line up with what I am seeing so the fix is easy.

Comment (EmilyBourne) [reply pyproject.toml]: 96221ceeb

Comment (yguclu) [reply pyproject.toml]: OK, yes. Please modify this file to use more explicit paths.

File: pyproject.toml [hunk 31]
```diff
@@ -48,7 +48,23 @@ path = "pyccel/version.py"
...
 
 [tool.hatch.build.targets.wheel.hooks.custom]
 path = "install_scripts/hook.py"
 require-runtime-dependencies = true
+
+
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyproject.toml]: 8ec5f1cc3

**Review by yguclu — APPROVED** [review]
